app/code/Magento/CatalogUrlRewrite/Observer/ProcessUrlRewriteOnChangeProductVisibilityObserver.php

```diff
@@ -15,6 +15,8 @@
 
 /**
  * Consider URL rewrites on change product visibility via mass action
+ *
+ * @deprecated
```

Comment:
Could you write some info why it was deprecated and what is/should be used instead? Such info will be really helpful during investigations